src/uct/ib/ud/base/ud_ep.h

```diff
@@ -153,15 +194,19 @@ struct uct_ud_ep {
          uct_ud_psn_t           psn;          /* Next PSN to send */
          uct_ud_psn_t           max_psn;      /* Largest PSN that can be sent - (ack_psn + window) (from incoming packet) */
          uct_ud_psn_t           acked_psn;    /* last psn that was acked by remote side */
-         uct_ud_psn_t           rt_psn;       /* last psn that was retransmitted */
          ucs_queue_head_t       window;       /* send window */
          uct_ud_ep_pending_op_t pending;      /* pending ops */
          ucs_time_t             send_time;    /* tx time of last packet */
-         ucs_queue_iter_t       rt_pos;       /* points to the part of tx window that needs to be resent */
          UCS_STATS_NODE_DECLARE(stats);
          UCT_UD_EP_HOOK_DECLARE(tx_hook);
     } tx;
     struct {
+         uct_ud_psn_t           psn;       /* last psn that was retransmitted */
+         uct_ud_psn_t           max_psn;   /* max psn that should be retransmitted */
+         ucs_queue_iter_t       pos;       /* points to the part of tx window that needs to be resent */
+         char                   is_active; /* resend operation is in progress */
```

Comment:
maybe better to declare this struct as packed. is it possible to eliminate some fields to reduce memory footprint?